core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -772,6 +772,12 @@ private[spark] class Executor(
     val accumUpdates = new ArrayBuffer[(Long, Seq[AccumulatorV2[_, _]])]()
     val curGCTime = computeTotalGcTime()
 
+    // get executor level memory metrics
+    val executorUpdates = new ExecutorMetrics(System.currentTimeMillis(),
+      ManagementFactory.getMemoryMXBean.getHeapMemoryUsage().getUsed(),
```

Comment:
What about including the jvms direct & memory mapped usage as well? see https://issues.apache.org/jira/browse/SPARK-22483

Reply:
It would be useful to have more information about offheap memory usage.

Reply:
We could add ManagementFactory.getMemoryMXBean.getNonHeapMemoryUsage().getUsed(), for total non-heap memory used by the JVM.

For direct and memory mapped usage, would collecting these be similar to https://gist.github.com/t3rmin4t0r/1a753ccdcfa8d111f07c ?

Reply:
to be honest I haven't tried this out myself, I only knew this was possible.  The version I was familiar with is whats used by the dropwizard metrics

https://github.com/dropwizard/metrics/blob/4.1-development/metrics-jvm/src/main/java/com/codahale/metrics/jvm/BufferPoolMetricSet.java

sorry you'll need to experiment with it a bit.  (again, not a blocker for this)

Reply:
Thanks, I will play around with it a bit. If it seems more complicated or expensive, I'll file a separate subtask.